lib/config-array/ignore-pattern.js

```diff
@@ -109,7 +109,7 @@ function dirSuffix(filePath) {
     return isDir ? "/" : "";
 }
 
-const DefaultPatterns = Object.freeze(["/**/node_modules/*"]);
+const DefaultPatterns = Object.freeze(["node_modules/"]);
```

Comment:
I"m pretty sure that you need the `**` above `node_modules` to match *nested* node_modules dirs. I agree the trailing `*` is not required, but the preceding `**` i'm pretty certain is.

Reply:
> If there is a separator at the beginning or middle (or both) of the pattern, then the pattern is relative to the directory level of the particular .gitignore file itself. Otherwise the pattern may also match at any level below the .gitignore level.
https://git-scm.com/docs/gitignore

Reply:
Sure, but according to that link the current version of this line should have identical behavior, no?

Since it doesn’t, eslint might not exactly match gitignore here, and that should be verified.

Reply:
On my local tests, both produced the same results. 
```
2021-06-19T02:41:24.741Z eslintrc:ignore-pattern Check {
  filePath: '/home/araujo/Área de trabalho/app-eurora.store/src/pages/node_modules/',
  dot: false,
  relativePath: 'src/pages/node_modules/',
  result: true
}
```

Reply:
Ah, so the log differences in the OP are illustrating that it’s more efficient, not different results?

Reply:
Exactly